src/EditorFeatures/Core.Wpf/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -3,41 +3,74 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.ErrorLogger;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Shared.Utilities;
 using Microsoft.VisualStudio.CodingConventions;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor.Options
 {
     // NOTE: this type depends Microsoft.VisualStudio.CodingConventions, so for now it's living in EditorFeatures.Wpf as that assembly
     // isn't yet available outside of Visual Studio.
     internal sealed partial class EditorConfigDocumentOptionsProvider : IDocumentOptionsProvider
     {
+        private const int EventDelayInMillisecond = 50;
+
         private readonly object _gate = new object();
+        private readonly object _eventGate = new object();
```

Comment:
having two gates here really means we need documentation on what each gate is covering and if there are any ordering constraints between the two.

Reply:
it is explained at the place it is used.

Reply:
I don't see anything in the usage that explains what the gate covers and if there is any ordering contraint between the two.  It would be helpful if htat information were placed here so that people could read this and understand what it applies to.